src/super_gradients/training/metrics/depth_estimation_metrics.py

```diff
@@ -0,0 +1,250 @@
+from typing import Tuple, Sequence, Union, Optional
+
+import torch
+from torch import Tensor
+from torchmetrics import MeanSquaredError, MeanSquaredLogError, MeanAbsoluteError, MeanAbsolutePercentageError, Metric
+from super_gradients.common.object_names import Metrics
+from super_gradients.common.registry import register_metric
+
+
+class DepthEstimationMetricMixin:
+    """
+    A mixin class providing common functionality for depth estimation metrics.
+
+    :param ignore_val: Value to be ignored when computing metrics.
+    :param apply_sigmoid: Whether to apply the sigmoid function to predictions before updating the metric.
+    """
+
+    def __init__(self, ignore_val: Optional[float] = None, apply_sigmoid: bool = False):
+        self.ignore_val = ignore_val
+        self.apply_sigmoid = apply_sigmoid
+
+    def process_preds_and_target(self, preds: Union[Tensor, Sequence[Tensor]], target: Tensor) -> Tuple[Tensor, Tensor]:
+        """
+        Process predictions and target tensors for depth estimation metrics:
+        - If a sequence is returned by the model -> sets preds to the first element
+        - Squeezes the dummy dimension (i.e dim1) when preds.ndim == 4
+        - Applies sigmoid to preds if apply_sigmoid is True
+        - Removes entries to ignore where ignore_val is present in target
+
+        :param preds: Model predictions, either a tensor or a sequence of tensors.
+        :param target: Ground truth depth map.
+        :return: Processed predictions and target tensors.
+        """
+
+        if isinstance(preds, Sequence):
+            preds = preds[0]
+
+        if preds.ndim == 4:
+            preds = preds.squeeze(1)
+
+        if self.apply_sigmoid:
+            preds = torch.sigmoid(preds)
+
+        if self.ignore_val is not None:
+            non_ignored = preds != self.ignore_val
+            preds = preds[non_ignored]
```

Comment:
I think it should be computed in a slightly different fashion.
The `preds = preds[non_ignored]` line would translate `[B,1,H,W]` tensor into `[N]` single-dimensional array which will be passed to metric and treated as a single element.

However the depth metrics usually compute the statistics per image and then average.
So what we should be doing instead is:
```
for pred_i, tgt_i in zip(preds, targets):
  if self.ignore_val is not None:
      non_ignored = tgt_i != self.ignore_val
      pred_i = pred_i[non_ignored]
      tgt_i = tgt_i[non_ignored]
  self.update(pred_i, tgt_i)
```

Reply:
Not true, It is mathemtically equivalent to take the mean over the pixels of each image (see formulas and go up the parents class you would see that there are two states being updated - total pixels, which is computed after flattening and extracting numeral() or shape[0] and the sum of squared errors).
So this is unnecessary.